lib/puppet/type/sensu_check.rb

```diff
@@ -69,6 +69,15 @@ def insync?(is)
     end
   end
 
+  newproperty(:contacts, :array_matching => :all, :parent => SensuCheckArrayProperty) do
+    desc "Contact names to override handler configuration via Contact Routing"
+    newvalues(/.*/, :absent)
```

Comment:
Regex has been added both to sensu_check's `contacts` property and sensu_contact's `name` parameter.  Spec tests have been updated to exercise the expected behavior of invalid names throwing Puppet::ResourceError's.